tests/baselines/reference/api/typescript.d.ts

```diff
@@ -9358,19 +9357,6 @@ declare namespace ts {
      * provided resolution mode in the reference, unless one is not present, in which case it is the mode of the containing file.
      */
     function getModeForFileReference(ref: FileReference | string, containingFileMode: ResolutionMode): ResolutionMode;
-    /**
-     * Use `program.getModeForResolutionAtIndex`, which retrieves the correct `compilerOptions`, instead of this function whenever possible.
-     * Calculates the final resolution mode for an import at some index within a file's `imports` list. This is the resolution mode
-     * explicitly provided via import attributes, if present, or the syntax the usage would have if emitted to JavaScript. In
-     * `--module node16` or `nodenext`, this may depend on the file's `impliedNodeFormat`. In `--module preserve`, it depends only on the
-     * input syntax of the reference. In other `module` modes, when overriding import attributes are not provided, this function returns
-     * `undefined`, as the result would have no impact on module resolution, emit, or type checking.
-     * @param file File to fetch the resolution mode within
-     * @param index Index into the file's complete resolution list to get the resolution of - this is a concatenation of the file's imports and module augmentations
-     * @param compilerOptions The compiler options for the program that owns the file. If the file belongs to a referenced project, the compiler options
-     * should be the options of the referenced project, not the referencing project.
-     */
-    function getModeForResolutionAtIndex(file: SourceFile, index: number, compilerOptions: CompilerOptions): ResolutionMode;
```

Comment:
The fact that this is changing seems like a bug, or maybe a misunderstanding; what I'm intending is that our public API does not change at all, and that unused code is only ever reported on things that are internal; is that possible?

Reply:
Yeah that's what I meant with not being sure about the `hereby baseline-accept` thing.

I can't see any other change made in this PR with an export named `protocol`

Reply:
One of the `getModeForResolutionAtIndex` overloads has an  `@internal` tag, so a bit of an exception I guess.

Reply:
Restored + added `@knipignore` to this specific export.

Yet I fail to see what caused the other changes in `api/typescript.d.ts`. Those exports aren't touched in this PR. Do you have an idea?

Reply:
I'll take a look. I'll probably force push this branch to attempt to clean this back up again after I'm done futzing with it. There are some incorrect things in the config that are I think leading to too much pruning and other bits.

Reply:
Going to wait to push until https://github.com/microsoft/TypeScript/pull/58229 just to make things easier later; my intent is to start from scratch with the config/etc you have, fix the test entrypoint, then apply the ignores where needed, removing the rest.

The output dts file is funky just because it's a bundled file created by `dtsBundler`; most likely what happened was that something stopped being referenced by an import, which then caused the bundler to not be able to find a specific import anymore, and then have to do something strange (which then required a private type and that caused export modifiers to show up).

Reply:
Thanks, that makes sense. Sounds like a plan, a few notes that may help here:

- You can use `--debug` to see what files are included (incl glob patterns etc)
- There's also `--trace` to see the individual export-import lookups (across re-exports etc)
- You may want to disable the `mocha` plugin to prevent it from adding entry files you may not expect/want (just do `mocha: false` in the config).
- Knip has a `--production` mode to exclude non-production (i.e. test) files which I think might be interesting for this repo, but unfortunately conflicts with `--tags=+internal` as Knip ignores `@internal` tags in production mode (I'll fix this so we can override). Either way, disabling the mocha plugin gives basically the same setup.